source/http_message.c

```diff
@@ -121,6 +121,10 @@ static void s_napi_http_request_finalize(napi_env env, void *finalize_data, void
     struct http_request_binding *binding = finalize_data;
     struct aws_allocator *allocator = finalize_hint;
 
+    if (binding->node_headers != NULL) {
+        napi_delete_reference(env, binding->node_headers);
```

Comment:
Leak Fix 1: The C HttpRequest binding was never releasing its reference to the JS HttpHeaders